LIMS2DB/classes.py

```diff
@@ -1042,3 +1043,63 @@ def extract_barcode(self, chain):
             if matches:
                 barcode = matches.group(0).replace('_','-')
         return barcode
+
+    def set_status(self):
+        proj_details = self.obj.get('details')
+        status_fields = {}
```

Comment:
Maybe all the status fields could be initialised to `False` here, so that you know what statuses are available and you can also cut down on the code setting them to false below.

Reply:
Ok, I think that made it better. What do you think?

Reply:
Would aborted be a tag or a status field? Since an aborted project can be closed.

Maybe we should be showing both in genstat. If a project is in `Needs Review`, we show it as 'Needs Review' only.

Reply:
I think these are overlapping statuses, the same as ongoing and open and reception control and open. Regarding Needs Review, I think they show up on ongoing, open and reception control as well where applicable.

Reply:
Ah yes, my bad, I was mixing them up in my mind.